src/Base/Tools.h

```diff
@@ -113,31 +113,53 @@ inline manipulator<int> blanks(int n)
 // ----------------------------------------------------------------------------
 
 template<class T>
+    requires std::is_arithmetic_v<T>
 inline T clamp(T num, T lower, T upper)
 {
     return std::max<T>(std::min<T>(upper, num), lower);
 }
 
-template<class T>
-inline T sgn(T t)
+/// Returns -1, 0 or 1 depending on if the value is negative, zero or positive

```

Comment:
Can we add comment here to include why it is implemented that way?:
```suggestion
/// Returns -1, 0 or 1 depending on if the value is negative, zero or positive
/// As this function is often used in hot paths, it uses branchless implementation
```

Reply:
I don't know if this is used in any hot path currently.
As there where a comment not to use a utility function (toRadians) as it would be slower than hard coding a radian angle, I preemptively optimized this to avoid similar ideas for this function.

Now it should be possible to use in a hot path, if needed.

Reply:
The comment simply anticipates such use as it was suggested that it is the case why it is not used there. Maybe "often" should be changed into "sometimes" though.

Reply:
I've now added a similar comment, what do you think?
```cpp
/// As this function might be used in hot paths, it uses branchless implementation
```

Reply:
Perfect!